src/tdywy.c

```diff
@@ -918,14 +926,16 @@ PetscErrorCode TDyWYResidual(TS ts,PetscReal t,Vec U,Vec U_t,Vec R,void *ctx) {
       sign = PetscSign(TDyADotBMinusC(&(tdy->N[dim*f]),&(tdy->X[dim*f]),
                                       &(tdy->X[dim*c]),dim));
       for(j=0; j<nv; j++) {
-        div += sign*tdy->vel[dim*(f-fStart)+j]*wgt*tdy->V[f];
+        div += sign*tdy->vel[nv*(f-fStart)+j]*wgt*tdy->V[f];
       }
     }
 
-    r[c] = tdy->porosity[c-cStart]*tdy->dS_dP[c-cStart]*dp_dt[c] + div;
-
+    r[c] = tdy->porosity[c-cStart]*tdy->dS_dP[c-cStart]*dp_dt[c] + div - tdy->Flocal[c-cStart];
+    //PetscPrintf(PETSC_COMM_WORLD,"R[%2d] = %+e %+e %+e = %+e\n",
+    // 	c,tdy->porosity[c-cStart]*tdy->dS_dP[c-cStart]*dp_dt[c],
+    // 		div,tdy->Flocal[c-cStart],r[c]);
```

Comment:
Consider deleting these lines.

Reply:
We aren't 'out of the woods' yet and I might still need to refer to them. Cleaning in a future commit.

Reply:
Okay.